shoov/modules/custom/shoov_restful/plugins/restful/node/ci_build_status/1.0/ShoovCiBuildStatusResource.class.php

```diff
@@ -70,17 +70,27 @@ protected function isValidEntity($op, $entity_id) {
     $params = array(
       '@id' => $entity_id,
     );
+    $node = node_load($entity_id);
+    $user = $this->getAccount();
+    if (!$entity_id) {
+      throw new RestfulNotFoundException(format_string('Invalid URL path'));
+    }
+
+    if ($node->type != 'ci_build') {
+      throw new RestfulUnprocessableEntityException(format_string('The entity ID @id is not a valid CI Build.', $params));
+    }
 
     $request = $this->getRequest();
+    if ($user && node_access('view', $node)) {
```

Comment:
There will always be an `$account` - you need to check `$account->uid`
also, you can move this line above the `$request`